### PR TITLE
Added Flutter Documentation extension

### PR DESCRIPTION
**Pull request by MattisBrizard**
## Description

This extension lets you search in the Flutter documentation and open the researched element on https://api.flutter.dev

## Screencast

https://user-images.githubusercontent.com/26067909/154865106-62dc1e9f-a94e-43a3-8ae4-a2e1ec0f5e66.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @MattisBrizard 💪 

Thanks for your contribution, it's always nice to get more documentation extensions 💯 

I left some feedback for you to address 😄 

Is there a way to show some default elements, like doing in the Helm documentation, I'm not sure if it's that easy with Flutter, just trying to see if there is a way to notch up the UX a bit 😄 

image

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by MattisBrizard** [comment]
Hi @pernielsentikaer, I've applied the suggestion you made and added an initial state showing more documentation links (Taken from https://api.flutter.dev/).
Tell me if you want me to add something else.

File: extensions/flutter-documentation-search/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+};
+
+export default function Command() {
+  const { state, search } = useSearch();
+
+  return (
+    <List
+      isLoading={state.isLoading}
+      onSearchTextChange={search}
+      searchBarPlaceholder="Search in ddze Flutter documentation..."
```
Comment (pernielsentikaer) [inline extensions/flutter-documentation-search/src/index.tsx:34]: What is `ddze` here?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @MattisBrizard 👋 

Only one question left, maybe I'm just too old to understand what it stands for 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by MattisBrizard** [comment]
Hi @pernielsentikaer, oops a typo, removed it 😅

**Comment by pernielsentikaer** [comment]
Ah thanks @MattisBrizard - I was feeling old for now knowing that 😄 

One last thing before merging, would it make sense to use the list.item icon in the searches (as the initial items have)

**Comment by MattisBrizard** [comment]
@pernielsentikaer, as some of the APIs shown are directly from the Dart SDK (Flutter embed the Dart SDK) showing a Flutter Icon for a Dart method wouldn't make sense for me.

**Comment by pernielsentikaer** [comment]
Got it, agree, thanks for the clarification 😄

**Review by pernielsentikaer — APPROVED** [review]
Approved! 💣

**Pull request merged**